tests/e2e/tests/test_rds_s3_irsa.py

```diff
@@ -0,0 +1,330 @@
+RDS_SECRET_PROVIDER_CLASS_FILE = (
+    f"{AWS_SECRETS_MANAGER_MANIFEST_FOLDER}/rds/secret-provider.yaml"
+)
+
+path_dic_rds_s3 = load_yaml_file(INSTALLATION_PATH_FILE)
+# pipelines helm path
+pipeline_rds_s3_helm_path = path_dic_rds_s3["kubeflow-pipelines"][
+    "installation_options"
+]["helm"]["paths"]
+
+# secrets-manager helm path
+secrets_manager_rds_s3_helm_path = path_dic_rds_s3["aws-secrets-manager"][
+    "installation_options"
+]["helm"]["paths"]
+
+# pipelines values file
+pipeline_rds_s3_values_file = f"{pipeline_rds_s3_helm_path}/values.yaml"
+
+# secrets-manager values file
+secrets_manager_rds_s3_values_file = f"{secrets_manager_rds_s3_helm_path}/values.yaml"
+
+
+METADB_NAME = "metadata_db"
+
+
+@pytest.fixture(scope="class")
+def installation_path():
+    return INSTALLATION_PATH_FILE
+
+@pytest.fixture(scope="class")
+def associate_oidc(cluster, region):
+    associate_iam_oidc_provider(cluster_name=cluster, region=region)
+
+@pytest.fixture(scope="class")
+def login():
+    return "user@example.com"
+
+
+@pytest.fixture(scope="class")
+def configure_manifests(cfn_stack, aws_secrets_driver, region, cluster):
+    iam_client = get_iam_client(region=region)
+
+    cfn_client = get_cfn_client(region)
+    stack_outputs = get_stack_outputs(cfn_client, cfn_stack["stack_name"])
+
+    rds_secret_provider = unmarshal_yaml(RDS_SECRET_PROVIDER_CLASS_FILE)
+    rds_secret_provider_objects = yaml.safe_load(
+        rds_secret_provider["spec"]["parameters"]["objects"]
+    )
+    rds_secret_provider_objects[0]["objectName"] = "-".join(
+        stack_outputs["RDSSecretName"].split(":")[-1].split("-")[:-1]
+    )
+    rds_secret_provider["spec"]["parameters"]["objects"] = yaml.dump(
+        rds_secret_provider_objects
+    )
+
+    with open(RDS_SECRET_PROVIDER_CLASS_FILE, "w") as file:
+        yaml.dump(rds_secret_provider, file)
+
+    rds_params = {
+        "dbHost": stack_outputs["RDSEndpoint"],
+        "mlmdDb": METADB_NAME,
+    }
+
+    s3_params = {
+        "bucketName": stack_outputs["S3BucketName"],
+        "minioServiceHost": "s3.amazonaws.com",
+        "minioServiceRegion": region,
+    }
+
+    configure_env_file(env_file_path=KFP_RDS_PARAMS_ENV_FILE, env_dict=rds_params)
+
+    configure_env_file(env_file_path=KFP_S3_PARAMS_ENV_FILE, env_dict=s3_params)
+
+    write_env_to_yaml(rds_params, pipeline_rds_s3_values_file, module="rds")
+    write_env_to_yaml(s3_params, pipeline_rds_s3_values_file, module="s3")
+
+    rds_secret_params = {
+        "secretName": "-".join(
+            stack_outputs["RDSSecretName"].split(":")[-1].split("-")[:-1]
+        )
+    }
+
+    write_env_to_yaml(
+        rds_secret_params, secrets_manager_rds_s3_values_file, module="rds"
+    )
+
+    os.environ["CLUSTER_REGION"] = region
+    os.environ["CLUSTER_NAME"] = cluster
+    apply_retcode = subprocess.call(f"make bootstrap-pipelines".split(), cwd="../..")
```

Comment:
Will the error be printed out if the call fails and is the iam role deleted upon uninstallation?

Reply:
it iwll print out if it fails, let me add a yield to run deletion.